skeleton/aws/modules/rds/main.tf

```diff
@@ -4,8 +4,8 @@ module "db" {
 
   name = "${var.namespace}-aurora-db"
 
-  engine         = var.engine
-  engine_version = var.engine_version
+  engine         = "aurora-postgresql"
+  engine_version = 15.3
```

Comment:
I also do not suggest hardcoding the values for these 2 settings because each project may use different engines (e.g. MySQL in easyHotel, PostgreSQL in Flexi Payroll, etc). These values should be customizable as the `core/variables.tf` instead of being a burden in this module. 
The reason is when bootstrapping a project, we tend to customize the values of the variables under `core/variables.tf` for all modules instead of looking at each sub modules.

Reply:
That makes sense to me 👍
---

Just a side note, regarding Terraform Cloud variables for application configurations, we're moving away from that due to some reasons:
- We are unable to audit the changes which can happen mistakenly
- The changes cannot be reviewed (through code review) before applying.

we will use `locals` instead, sth like (it was applied in another project)
```hcl
current_db_config = local.db_config[var.environment]
db_config = {
  staging = {
    rds_instance_type = "db.t3.small"
  }

  prod = {
    rds_instance_type  = "db.t3.large"
  }
}
```

we will use locals to manage application env instead (I'll work on that in a separate PR)

Reply:
Thanks for your insights, good to hear that 🚀

Reply:
My concern is that we sometime inline values and at other times use a `locals` block. That introduces inconsistencies. Shouldn't we agree to always use locals and place it at at the top of the `main.tf` file?

```
locals {
  engine                = "aurora-postgresql"
  engine_version = 15.3
}


module "db" {
  engine                = local.engine
  engine_version = local.engine_version

}
```

In addition to consistency, having locals at the top of the file make it easier to maintain and discover.

Reply:
At first glimpse, I intended to use more locals as this is indeed what we are used to do in other stacks. 🤔 
But the Terraform documentation [does not recommend it](https://developer.hashicorp.com/terraform/language/values/locals#when-to-use-local-values:~:text=Use%20local%20values%20only%20in%20moderation%2C%20in%20situations%20where%20a%20single%20value%20or%20result%20is%20used%20in%20many%20places%20and%20that%20value%20is%20likely%20to%20be%20changed%20in%20future.%20The%20ability%20to%20easily%20change%20the%20value%20in%20a%20central%20place%20is%20the%20key%20advantage%20of%20local%20values.) 👀 

> Use local values only in moderation, in situations where a single value or result is used in many places and that value is likely to be changed in the future. The ability to easily change the value in a central place is the key advantage of local values.

I think it's due to the declarative nature of TF: you want to keep values declared where they are used. So if your file gets bigger (ex need 2 RDS with different engines), you won't introduce too many abstractions with locals).

This is of course open for discussion and we are free to not follow strictly the recommendations of the documentation if we have a good reason to.

Reply:
Understood. Unless there is a massive downside, code-style wise I would prefer to have more consistency but indeed it requires a team discussion.

Reply:
Yes, we should have an RFC to pick the code style for this. But IMO, for some values that we can change for each project, we should add them into the locals, then developers can easily change them instead of finding them in the resources to modify.
But we can make that change later once we collect the votes